src/lambdas/user-update.ts

```diff
@@ -0,0 +1,55 @@
+import * as AWS from 'aws-sdk'
+
+const TABLE_NAME = process.env.TABLE_NAME || ''
+const PRIMARY_KEY = process.env.PRIMARY_KEY || ''
+
+const RESERVED_RESPONSE = `Error: You're using AWS reserved keywords as attributes`,
+  DYNAMODB_EXECUTION_ERROR = `Error: Execution update, caused a Dynamodb error, please take a look at your CloudWatch Logs.`
+
+const db = new AWS.DynamoDB.DocumentClient()
+
+export const handler = async (event: any = {}): Promise<any> => {
+  const editedUser: any =
+    typeof event.arguments?.user == 'object' ? event.arguments.user : JSON.parse(event.arguments.user)
+
+  const userId = event.arguments?.user?.id
+  if (!userId) {
+    return { statusCode: 400, body: 'invalid request, you are missing the user.id argument' }
+  }
+
+  delete editedUser.id
+  editedUser['updatedAt'] = new Date().toISOString()
+  let editedUserProps = Object.keys(editedUser)
+  if (!editedUser || editedUserProps.length < 2) {
+    return { statusCode: 400, body: 'invalid request, no arguments provided' }
+  }
+
+  const firstProperty = editedUserProps.splice(0, 1)
+  const params: any = {
+    TableName: TABLE_NAME,
+    Key: {
+      [PRIMARY_KEY]: userId,
+    },
+    UpdateExpression: `set ${firstProperty} = :${firstProperty}`,
+    ExpressionAttributeValues: {},
+    ReturnValues: 'UPDATED_NEW',
+  }
+  params.ExpressionAttributeValues[`:${firstProperty}`] = editedUser[`${firstProperty}`]
+
+  editedUserProps.forEach((property) => {
+    params.UpdateExpression += `, ${property} = :${property}`
+    params.ExpressionAttributeValues[`:${property}`] = editedUser[property]
+  })
+
+  try {
+    await db.update(params).promise()
```

Comment:
How would you encapsulate the calls to dynamodb, so in a way that if in the future we want to change the database it won't affect the handler logic?

Reply:
I would abstract away the dynamodb functionality into a single function. That way if we want to switch databases we can just change one line in the handler. And I would probably go a step further. This code to update an item in DynamoDB is pretty much generic. It can work for any entity/properties so I would probably encapsulate in its own function for reusability.